jerry-core/ecma/builtin-objects/ecma-builtin-bigint.c

```diff
@@ -36,6 +52,358 @@
  * @{
  */
 
+/**
+ * The BigInt object's 'asIntN' and 'asUintN' routines
+ *
+ * See also:
+ *          ECMA-262 v5, 11.0
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_bigint_object_as_int_n (ecma_value_t this_arg, /**< 'this' argument */
+                                     ecma_value_t bits, /**< number of bits */
+                                     ecma_value_t bigint, /**< bigint number */
+                                     bool is_signed) /**< The operation is signed */
+{
+  JERRY_UNUSED (this_arg);
+
+  ecma_number_t input_bits;
+  ecma_value_t bit_value = ecma_op_to_index (bits, &input_bits);
+
+  if (ECMA_IS_VALUE_ERROR (bit_value))
+  {
+    return bit_value;
+  }
+
+  ecma_value_t bigint_value = ecma_bigint_to_bigint (bigint, false);
+
+  if (ECMA_IS_VALUE_ERROR (bigint_value))
+  {
+    return bigint_value;
+  }
+
+  if (input_bits == 0 || bigint_value == ECMA_BIGINT_ZERO)
+  {
+    ecma_free_value (bigint_value);
+    return ECMA_BIGINT_ZERO;
+  }
+
+  const uint32_t four_byte = (sizeof (uint32_t) * sizeof (uint64_t));
+  const uint32_t one_byte = sizeof (uint64_t);
+
+  uint32_t whole_part = (ecma_value_t) input_bits / four_byte;
+  uint32_t remainder = (ecma_value_t) input_bits % four_byte;
+
+  uint32_t input_bit_length = 1;
+
+  if (whole_part == 0)
+  {
+    input_bit_length = (remainder > 0) ? (ecma_value_t) input_bits : four_byte;
+  }
+  else
+  {
+    input_bit_length = (remainder > 0) ? (uint32_t) (whole_part + 1) * four_byte : (uint32_t) (whole_part * four_byte);
+  }
+
+  uint32_t input_byte_size = (ecma_value_t) input_bits / one_byte;
+
+  if ((ecma_value_t) input_bits % one_byte != 0)
+  {
+    input_byte_size += 1;
+  }
+
+  ecma_extended_primitive_t * input_bigint_p = ecma_get_extended_primitive_from_value (bigint_value);
+
+  if (JERRY_UNLIKELY (input_bigint_p == NULL))
+  {
+    return 0;
+  }
+
+  uint32_t bigint_size = ECMA_BIGINT_GET_SIZE (input_bigint_p);
+  uint8_t input_bigint_sign = input_bigint_p->u.bigint_sign_and_size & ECMA_BIGINT_SIGN;
+
+  uint32_t min_size = ((input_bit_length / one_byte) < bigint_size) ? (input_bit_length / one_byte) : bigint_size;
```

Comment:
`input_bit_length / one_byte` used many times, maybe we could put it into a local valriable. also this could be a `>>` shift